Source/Scene/TimeDynamicPointCloud.js

```diff
@@ -0,0 +1,740 @@
+    }
+
+    function setFramesDirty(that, clippingPlanesDirty, styleDirty) {
+        var frames = that._frames;
+        var framesLength = frames.length;
+        for (var i = 0; i < framesLength; ++i) {
+            var frame = frames[i];
+            if (defined(frame) && defined(frame.pointCloud)) {
+                frame.pointCloud.clippingPlanesDirty = clippingPlanesDirty;
+                frame.pointCloud.styleDirty = styleDirty;
+            }
+        }
+    }
+
+    var updateState = {
+        timeSinceLoad : 0,
+        isClipped : false,
+        clippingPlanesDirty : false
+    };
+
+    /**
+     * Called when {@link Viewer} or {@link CesiumWidget} render the scene to
+     * get the draw commands needed to render this primitive.
+     * <p>
+     * Do not call this function directly.  This is documented just to
+     * list the exceptions that may be propagated when the scene is rendered:
+     * </p>
+     */
+    TimeDynamicPointCloud.prototype.update = function(frameState) {
+        if (frameState.mode === SceneMode.MORPHING) {
+            return;
+        }
+
+        if (!this.show) {
+            return;
+        }
+
+        if (!defined(this._pickId)) {
+            this._pickId = frameState.context.createPickId({
+                primitive : this
+            });
+        }
+
+        if (!defined(this._loadTimestamp)) {
+            this._loadTimestamp = JulianDate.clone(frameState.time);
+        }
+
+        // For styling
+        var timeSinceLoad = Math.max(JulianDate.secondsDifference(frameState.time, this._loadTimestamp) * 1000, 0.0);
+
+        // Update clipping planes
+        var clippingPlanes = this._clippingPlanes;
+        var clippingPlanesState = 0;
+        var clippingPlanesDirty = false;
+        var isClipped = defined(clippingPlanes) && clippingPlanes.enabled;
+
+        if (isClipped) {
+            clippingPlanes.update(frameState);
+            clippingPlanesState = clippingPlanes.clippingPlanesState;
+        }
+
+        if (this._clippingPlanesState !== clippingPlanesState) {
+            this._clippingPlanesState = clippingPlanesState;
+            clippingPlanesDirty = true;
+        }
+
+        var styleDirty = this._styleDirty;
+        this._styleDirty = false;
+
+        if (clippingPlanesDirty || styleDirty) {
+            setFramesDirty(this, clippingPlanesDirty, styleDirty);
+        }
+
+        updateState.timeSinceLoad = timeSinceLoad;
+        updateState.isClipped = isClipped;
+
+        var shading = this.shading;
+        var eyeDomeLighting = this._pointCloudEyeDomeLighting;
+
+        var commandList = frameState.commandList;
+        var lengthBeforeUpdate = commandList.length;
+
+        var previousInterval = this._previousInterval;
+        var nextInterval = this._nextInterval;
+        var currentInterval = getCurrentInterval(this);
+
+        if (!defined(currentInterval)) {
+            return;
+        }
+
+        var clockMultiplierChanged = false;
```

Comment:
You may have already considered this, but would it make more sense to handle the current/previous/next interval logic in a function that subscribes to the clock tick event instead of in the render loop itself? This is what `TimeDynamicImagery` does.